src/VideoReceiver/GstVideoReceiver.h

```diff
@@ -103,7 +103,7 @@ protected slots:
 
 protected:
     virtual GstElement* _makeSource(const QString& uri);
-    virtual GstElement* _makeDecoder(GstCaps* caps, GstElement* videoSink);
+    virtual GstElement* _makeDecoder();
```

Comment:
You changed class API in a way that breaks derived classes. The fact that you don't see them does not mean that they do no exist - be careful with `virtual`.

Perhaps it would be useful to look into a year old discussion regarding video refactoring - it explains a lot.

Reply:
Ok, I see what are you talking about, I'll change it back.

Reply:
> Moving to decodebin3 and enabling the hardware decode creates a much better scenario with a modern decodebin alternative that can handle a better construction and decision about the pipeline and elements that is used.

@patrickelectric can you please show better construction and decision about the pipeline and elements that are used?

Also, are you sure that decodebin3 works on Android and iOS (in the way how do you use it)? I recall that these platforms were not so straightforward.

Reply:
decodebin failed to use nvidia hardware decoders and create the correct filtercaps for glupload
I also got similar problems with some vaapi decoders. 

I have not tested with android yet.

Reply:
Android works without problem.